packages/gradients/src/GradientFactory.ts

```diff
@@ -11,14 +11,26 @@ import type { Renderer } from '@pixi/core';
  * @ignore
  * @param color - The hexadecimal form of the color.
  */
-function cssColor(color: number) {
+function cssColor(color: number | string) {
+    if(typeof color === 'string'){
+        return color;
+    }
     let string = color.toString(16);
 
-    while (string.length < 6) {
-        string = `0${string}`;
+    if(string.length === 3){
+        string = `#${string[0]}${string[0]}${string[1]}${string[1]}${string[2]}${string[2]}`;
+    }
```

Comment:
This actually doesn't work. You can't distinguish if `0xABC` means `#AABBCC` or `#000ABC`.

Reply:
@ShukantPal, would anyone actually use `0xABC` meaning `#000ABC`? In CSS, if you were to specify a color as `#ABC`, it is used as though it is `#AABBCC`.